R/createFigure4.R

```diff
@@ -0,0 +1,220 @@
+      chunk.call = do.call(readWQPdata, params)
+    
+      if(!is.null(chunk.call)){
+        meta <- attr(chunk.call, "siteInfo") %>%
+          select(MonitoringLocationIdentifier, dec_lat_va, dec_lon_va, StateCode)
+        
+        chunk.call <- rename(chunk.call, 
+                             Date=ActivityStartDate, 
+                             value=ResultMeasureValue, 
+                             units=ResultMeasure.MeasureUnitCode) %>% 
+          left_join(meta, by="MonitoringLocationIdentifier") %>%
+          rename(wqx.id=MonitoringLocationIdentifier) %>%
+          mutate(StateCode = as.character(StateCode),
+                 ActivityTypeCode = as.character(ActivityTypeCode),
+                 ActivityMediaName = as.character(ActivityMediaName),
+                 ResultStatusIdentifier = as.character(ResultStatusIdentifier)) %>%
+          select(Date, value, units, wqx.id, 
+                 dec_lat_va, dec_lon_va, 
+                 StateCode, 
+                 ActivityTypeCode,ActivityMediaName,
+                 ResultStatusIdentifier)
+        
+        out[[i]] <- chunk.call       
+      }
+    }
+  }
+  
+  if(length(out) > 0){
+    out.data <- bind_rows(out)
+    
+  } else {
+    out.data <- data.frame(Date = numeric(),
+                           value=numeric(),
+                           units = character(),
+                           wqx.id = character(),
+                           dec_lat_va = numeric(),
+                           dec_lon_va = numeric(),
+                           StateCode = character(),
+                           ActivityTypeCode = character(),
+                           ActivityMediaName = character(),
+                           ResultStatusIdentifier = character())
+  }
+  
+  return(out.data)
+
+}
+
+
+
+library(dplyr)
+library(dataRetrieval)
+library(lubridate)
+library(ggplot2)
+
+characteristicNames = c("Depth, Secchi disk depth", 
+                        "Depth, Secchi disk depth (choice list)", 
+                        "Secchi Reading Condition (choice list)", 
+                        "Secchi depth", 
+                        "Water transparency, Secchi disc")
+
+# To get the full data set used to produce the figure in the text:
+# This takes roughly 15 minutes
+get_us_secchi(outfile="all_secchi_usa_long.rds",
+              characteristicNames,
+              siteTypes = "All",
+              stateCode = "All",
+              startDateLo = '1900-01-01',
+              startDateHi = '2016-01-01',
+              stride = "20 years")
+
+# To get the small subset to test the workflow:
+# This takes roughly 3 minutes
+get_us_secchi(outfile="sub_secchi_AL_MN.rds",
+              characteristicNames,
+              siteTypes = "Lake, Reservoir, Impoundment",
+              stateCode = c("01","27"), #AL and MN
+              startDateLo = '2000-01-01', 
+              startDateHi = '2016-01-01',
+              stride = "20 years")
+
+infile <- "all_secchi_usa_long.rds"
```

Comment:
ah - it's because this should needs to be changed based on which one of the get_us_secchi commands I ran. Could you just add a note about this matching the outfile in the command they used?